readme/ByteParsers.scalatex

```diff
@@ -177,6 +178,81 @@
+                try one of these:
+
+            @ul
+                @for(name <- Seq("canon.mid", "go.mid", "tonghua.mid"))
+                    @li
+                        @lnk(name, s"https://raw.githubusercontent.com/lihaoyi/fastparse/master/fastparse/jvm/src/test/resources/$name")
+
+            @p
+                If you wish, you can also check out this project and run
+                @code{sbt "fastparseJVM/test:run go.mid"} from the command line
+                to play the same MIDI file using the same parser, but running on
+                Scala-JVM and using the JVM's MIDI player rather than the browser's.
+                In general, the JVM player is much more robust than the Web player:
+                the Web player doesn't know how to handle changes in pitch bend,
+                polyphonic pressure, channel pressure or controller, and plays
+                all notes as a single piano-like instrument
+
+            @p
+                MidiParse comprises about 50 lines of @hl.scala{case class}es that
+                define the structure of the midi file, and another 140 lines of
+                FastParse code for parsing the MIDI from binary input:
+
+            @ul
+                @li
+                    @a("MidiParse Source", href:="https://github.com/lihaoyi/fastparse/tree/master/fastparse/shared/src/test/scala/fastparse/MidiParse.scala")
+
+            @p
+                MidiParse illustrates many of the quirks of writing a parser for
+                a real-world binary format:
+
+            @ul
+                @li
+                    How to parse various sorts of bit-packed integers: 8-bit,
+                    16-bit, or 32-bit signed and unsigned integers (e.g. @code{UInt8}).
+                    Even more obscure things like the 24-bit integer used to
+                    store the @code{Tempo} of the music, or the variable-length
+                    @code{varInt} parser keeps consuming individual bytes
+                    until it finds one whose high-order bit is set, and then
+                    concats them all into one variable-length integer.
+                @li
+                    @code{varString} and @code{varByte} parsers are a common
+                    case of using @sect.ref{FlatMap} to parse a variable-length
+                    data-field whose length is defined up-front
+                @li
+                    @code{midiEvent} and @code{metaEvent} show how to switch
+                    on a single byte upfront and use that to decide how to
+                    continue parsing a particular item, using
+                    @sect.ref{FlatMap}/@hl.scala{for}-comprehensions
+                @li
+                    How to use @sect.ref{Filter} to narrow down the sort of
+                    input a parser can parse, e.g. to narrow a
+                    @hl.scala{Int8: Parser[Byte]} into a parser that only
+                    accepts non-negative bytes
```

Comment:
why doesn't midi format pack non-negative bytes into 7 bits considering all this mess with `varInt` and 24bits? :D